deep_qa/data/tokenizers/word_tokenizer.py

```diff
@@ -19,7 +20,8 @@ class WordTokenizer(Tokenizer):
     """
     def __init__(self, params: Dict[str, Any]):
         word_splitter_choice = get_choice_with_default(params, 'word_splitter', list(word_splitters.keys()))
-        self.word_splitter = word_splitters[word_splitter_choice]()
+        word_processor_choice = get_choice_with_default(params, 'word_processor', list(word_processors.keys()))
```

Comment:
I think the API would be cleaner and easier to extend if the parameters we passed looked something like this:

```json
{
  "processor": {
    "splitter": {
    ...
    },
    "stemmer": {
    ...
    },
    "stopwords": {
    ...
    }
  }
}
```

The tokenizer then just has one object that it constructs (the `WordProcessor`), and it only interacts directly with that object, with the single function call `get_tokens(text: str) -> List[str]`.  The `WordProcessor` then handles splitting words and whatever stemming or stop word removal you've specified in your parameters.  Does that seem reasonable?